Support server-side auth

benediktkaiser opened the pull request:
closes #35

benediktkaiser commented:
I added the private wallet key that we will be using for development purposes in `auction-ui/dev` with the key `PRIVATE_WALLET_KEY`. It can also be found on Bitwarden. 

@florianstoecker will add some ETH funds on the test networks that we can test the server side wallet functions.

valiafetisov commented:
> I added the private wallet key

Please note that we changed how we organise secrets in the last sprint. Now they are organised in `project-name/host/service` format; in other words, the correct place to which you need to add secret is `auction-ui/localhost/bot-twitter`. I would also recommend to always start your environment variable name with the most important word, in other words name it `WALLET_PRIVATE_KEY` instead. If we would have `WALLET_ADDRESS` in the future (and we may already add it there, since we need this information in the future) they will be conceptually grouped together.

benediktkaiser commented:
> * [ ]  Please note that we changed how we organise secrets in the last sprint. Now they are organised in `project-name/host/service` format; in other words, the correct place to which you need to add secret is `auction-ui/localhost/bot-twitter`

Thank you I will add it there now!


> * [ ]  I would also recommend to always start your environment variable name with the most important word, in other words name it `WALLET_PRIVATE_KEY` instead. If we would have `WALLET_ADDRESS` in the future (and you should already add it there) they will be conceptually grouped together

Makes sense! I will do that!

benediktkaiser commented:
I have changed the naming of the env variable! 

I wanted to ask how to best test if the `private_key` is working correctly. I had a look into the tests of DAI.js, but noticed that they have commented out all of their tests that involve a private wallet key. 

https://github.com/makerdao/dai.js/blob/dev/packages/dai/test/Maker.spec.js#L32

@valiafetisov do you have an idea for what kind of function I could write to test the implementation? Until now I have only been able to `console.log` the maker object and it includes the private key, however an actual test to see if the wallet can actually be used would be good to have!

benediktkaiser commented:
> * [ ]  Please note that we changed how we organise secrets in the last sprint. Now they are organised in `project-name/host/service` format; in other words, the correct place to which you need to add secret is `auction-ui/localhost/bot-twitter`

Also quick note to this: I do not have enough permission to write to these secrets. Who could I ask to give me permissions?

valiafetisov commented:
> what kind of function I could write to test the implementation?

They issue specifies which function you should call: authorization. More specifically, it suggests to check auth status (ie `getWalletAuthorizationStatus`) and then execute it (ie via `authorizeWallet`).

benediktkaiser commented:
I have incorporated the authorisation check into the loop, however I am getting no feedback from the function. I also added console.log at multiple breakpoints, but the code is not being executed. 

Do you maybe see a flaw in my approach?

valiafetisov commented on core/src/maker.ts:
```diff
@@ -26,6 +27,7 @@ const createMaker = async function (network: string): Promise<typeof Maker> {
             url: networkConfig.url,
             type: 'HTTP',
         },
+        privateKey,
```
As I suggested, and you pointed out in the comment above, this may not work and before debugging our code, please validate that passing `privateKey` is actually properly handled by the dai.js library. You may find this in the dai.js docs, in the code itself or in the test. It may be that you also need to pass something else or at least call `maker.useAccountWithAddress(address)`

benediktkaiser replied:
I had assumed as you sent me the following code snippet, but I will look further into it!

benediktkaiser commented:
Aside from trying to pass the `privateKey` I am having issues executing other functions in the bot in general. 

```js
(await getNewAuctions()).map(notify);
console.log("TEST")
await checkWalletAuthorization();
```

I would expect the following code to result in "TEST" being printed in the console, however this does not happen. This is a test on `bot-twitter/src/index.ts`. So I am also confused why simple console logs are not being printed correctly.

valiafetisov commented:
Then again, the first thing that you should investigate is: "what happens when I pass `privateKey` to `Maker.create`?".

benediktkaiser commented:
> Then again, the first thing that you should investigate is: "what happens when I pass `privateKey` to `Maker.create`?".

I found references to the `privateKey` in the Maker Docs as well. Therefore I still assume that this is the right way to do this.

valiafetisov commented:
> I found references to the privateKey in the Maker Docs as well

Please document them here or in the issue. As documentation can be outdated, you should also have a look at the source code of the dai.js version we're using and also document/outline what's going on there

benediktkaiser commented:
> > I found references to the privateKey in the Maker Docs as well
> 
> Please document them here or in the issue. As documentation can be outdated, you should also have a look at the source code of the dai.js version we're using and also document/outline what's going on there

I thought I had added the link. (https://docs.makerdao.com/build/dai.js/getting-started#create-a-vault). As edited in the previous comment, when searching through Dai.JS you can see that it is used in their tests and source:

https://github.com/makerdao/dai.js/search?p=1&q=privateKey

benediktkaiser commented:
As discussed in the daily on the 25.1.22, this PR we no longer be required as we are working on removing DaiJS from our project anyway. Therefore proposing a solution to use server side auth with DaiJS is no longer required.
